app/backends/workflow-orchestration.ts

```diff
@@ -81,23 +84,91 @@ async function _handleWorkItemResults(
 }
 
 /**
- * Creates a work item that uses all the output of the previous step.
+ * Read a STAC catalog and return the item links. This does not handle sub-catalogs. This function 
+ * makes assumptions based on the Harmony STAC directory layout for services inputs/outputs and
+ * is only intended to be used when aggregating service outputs into a single catalog.
+ * @param catlogPath - the path to the catalog
+ */
+async function getLinksFromCatalog(catalogPath: string): Promise<StacItemLink[]> {
+  const baseDir = path.dirname(catalogPath).replace(env.hostVolumePath, PATH_TO_CONTAINER_ARTIFACTS);
+  const text = (await fs.readFile(catalogPath)).toString();
+  const catalog = JSON.parse(text);
+  const links: StacItemLink[] = [];
+  for (const link of catalog.links) {
+    if (link.rel === 'item') {
+      // make relative path absolute
+      const { href } = link;
+      link.href = `${baseDir}/${path.normalize(href)}`;
+      links.push(link);
+    }
+  }
+
+  return links;
+}
+
+/**
+ * Creates a work item that uses all the output of the previous step. This function assumes that
+ * all work items for the previous step are completed. It also relies on the convention that
+ * services write out their results as STAC catalogs with the following path
+ * `/tmp/<JOB_ID>/<WORK_ITEM_ID>/outputs /catalogN.json`
+ * where N is from 0 to the number of results - 1.
  * 
  * @param tx - The database transaction
  * @param currentWorkItem - The current work item
  * @param nextStep - the next step in the workflow
- * @param results - an array of paths to STAC catalogs
+ * @param results - an array of paths to STAC catalogs from the last worked item
  */
 async function createAggregatingWorkItem(
-  tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep, results: string[],
+  tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep,
 ): Promise<void> {
-  // TODO use ALL outputs from the prior step here (HARMONY-961)
-  // just use the first result of the current output for now
+  const itemLinks: StacItemLink[] = [];
+  // get all the previous results
+  const workItemCount = await workItemCountForStep(tx, currentWorkItem.jobID, nextStep.stepIndex - 1);
+  let page = 1;
+  let processedItemCount = 0;
+  while (processedItemCount != workItemCount) {
+    const prevStepWorkItems = await getWorkItemsByJobIdAndStepIndex(tx, currentWorkItem.jobID, nextStep.stepIndex - 1, page);
+    for (const workItem of prevStepWorkItems.workItems) {
```

Comment:
Having to read all of the catalogs puts a some kind of upper limit on the number of granules we can reasonably handle for an aggregation step. Something we should keep in mind for the no granule limits epic.

Reply:
Yes, as I have mentioned before just using STAC to communicate between steps is limiting since JSON must be entirely read into memory to be parsed. We should think about using a simpler format that can be read (and parsed) line by line.

Reply:
Or page-by-page, with one catalog per page

Reply:
So on that note, this is 1000% going to come up with the rest of no granule limit, and we've already done the paging parts of that multi-epic.  Given this introduces a paging issue, I think we at least need a ticket to tackle it next PI, if not entirely handling the paging now